partition_gpu/partition_gpu.go

```diff
@@ -169,6 +187,24 @@ func enableMigMode() error {
 	return exec.Command(*nvidiaSmiPath, "-mig", "1").Run()
 }
 
+// checkGpuType checkes the GPU type used
+func checkGpuType() (string, error) {
+	gpuType, err := exec.Command(*nvidiaSmiPath, "--query-gpu=gpu_name", "--format=csv,noheader").Output()
+	if err != nil {
+		return "", err
+	}
+	if strings.HasPrefix(string(gpuType), Nvidia80gbH100) {
```

Comment:
Can we use switch here instead of if..else...?

Reply:
Done